src/martian/directive.py

```diff
@@ -246,8 +249,8 @@ def check_factory_signature(self, *arguments, **kw):
         argspec = inspect.formatargspec(args[1:], varargs, varkw, defaults)
         exec("def signature_checker" + argspec + ": pass")
         try:
-            signature_checker(*arguments, **kw)
-        except TypeError, e:
+            locals()['signature_checker'](*arguments, **kw)
```

Comment:
This confuses me too! 
Without the locals() i get this error in python 3: NameError: name 'signature_checker' is not defined
With my last commit the code is using a namespace, so its a little bit nicer...
I wonder if possible to work without dynamic code?